### PR TITLE
Feature: Result Tuple

### PR DESCRIPTION
**Pull request by asleepace**
# Description

This is a fairly large update which expands the result tuple functionality, includes more utility methods, expands the test suite and improves documentation.

- new `Res` class which extends result tuple
- new `vet(fn)` shorthand utility
- add `.unwrap()`, `.unwrapOr()` and `.or(fn)` utility methods
- add large test suite for edge cases

This is planned to be released as `v0.2.0`.

### CODE REVIEW

**Review by asleepace — COMMENTED** [review]
LGTM!

File: README.md [hunk 1]
```diff
@@ -11,71 +11,134 @@ if (!error) {
...
+  console.log(link.value.href)
+} else {
+  console.warn(link.error.message)
+}
+
+// and ergonomic value-error-tuple (vet) shorthand:
+
+const link = vet(() => new URL('asleepace.com'))
+  .or(() => new URL('https://aslee pace.com'))
+  .or(() => new URL('https://github.com'))
```
[the remark this answers is not in the document]
Comment (asleepace) [reply README.md:36]: This is meant to be more visibly clear in the demo.
